src/Symfony/Bundle/FrameworkBundle/Test/TestContainer.php

```diff
@@ -69,7 +68,16 @@ public function setParameter(string $name, mixed $value)
 
     public function set(string $id, mixed $service)
     {
-        $this->getPublicContainer()->set($id, $service);
+        $renamedId = $this->renamedIds[$id] ?? $id;
+
+        try {
+            $this->getPublicContainer()->set($renamedId, $service);
+        } catch (InvalidArgumentException $e) {
+            if (!str_starts_with($e->getMessage(), "The \"$renamedId\" service is private")) {
+                throw $e;
+            }
+            $this->getPublicContainer()->privates[$renamedId] = $service;
```

Comment:
should we throw if the private service was already instantiated, to avoid WTF debugging about why the new instance being set is not used by other services ? This would be consistent with the error throw in Container::set when trying to replace a public service that has already been instantiated.

Reply:
yup! added!